### ISSUE TITLE
Pandoc bad export from md to typst (#horizontalrule error)

### ISSUE THREAD

**Issue by Disonantemus** [jgm/pandoc#9506]
Pandoc choke with himself. Steps:
1. Create a simple markdown with two words and a ruler in between (echo).
2. Convert to typst.
3. Convert to markdown and show in stdout.

Used commands and output (error):
~~~bash
$ echo -e "one\n\n---\n\ntwo" | pandoc - -t typst | pandoc - -f typst -t markdown
(line 3, column 2):
Identifier "horizontalrule" not found
~~~

Same previous steps without Nº3, output redirected to a file and show content:
~~~bash
$ echo -e "one\n\n---\n\ntwo" | pandoc - -t typst > test.typ
$ cat test.typ
one

#horizontalrule

two
~~~

Convert typst to markdown (error):
~~~bash
$ pandoc test.typ -t markdown
"test.typ" (line 3, column 2):
Identifier "horizontalrule" not found
~~~

Convert typst to pdf (using typst, error):
~~~bash
$ typst compile test.typ
error: unknown variable: horizontalrule
  ┌─ test.typ:3:1
  │
3 │ #horizontalrule
  │  ^^^^^^^^^^^^^^
~~~

I did a search in google about: **typst** "#horizontalrule" with no hits, also search in [typst.app](https://typst.app/docs). Here in GH: Issues, Discussions and the whole repo. With all the info here, I believe **there is a bug** in that conversion.

The most similar thing that I found to [Horizontal Rules](https://pandoc.org/MANUAL.html#horizontal-rules) in **typst** was [this](https://typst.app/docs/reference/visualize/line/):
~~~c
#line(length: 100%)
~~~

**Workaround**: changing problematic line with sed fix errors
~~~bash
$ echo -e "one\n\n---\n\ntwo" \
| pandoc - -f markdown -t typst \
| sed 's/#horizontalrule/#line(length: 100%)/g' \
| pandoc - -f typst -t markdown
one

------------------------------------------------------------------------

two
~~~
And shows expected output.

---

**Note** : For _some strange reason,_ if I do convert, from **markdown** to **pdf** using **typst** engine, _works without error,_ and pdf output is not wrong:
~~~bash
$ echo -e "one\n\n---\n\ntwo" | pandoc - -t typst -o test.pdf
~~~
PDF output is 2 words to the left with a small (half-width length) _centered horizontal line_ in between.

---

Software versions (system was updated today, [pandoc-bin](https://aur.archlinux.org/packages/pandoc-bin))
~~~yaml
$ pandoc -v | head -3
pandoc 3.1.11
Features: +server +lua
Scripting engine: Lua 5.4

$ neofetch --off distro kernel
distro: Arch Linux x86_64
kernel: 6.1.52-1-lts
~~~

**Comment by jgm** [comment]
`horizontalrule` isn't a standard typst command. It's a function we define in the default typst template.
Use `-s` when converting to typst and you'll get this definition.

**Comment by Disonantemus** [comment]
I tested it and works, but does not mean that is right.

When not using `--standalone` flag, should be one of the following:
- Should be converted to **typst**.
- That _definition_ should be there.

Now I know, that, if I want to use my _custom pre formatting_, I need to:
- Convert **Pandoc**'s output (like my sed example).
- Add **Pandoc**'s definition in my _custom pre formatting_.

But should not be that way, to my knowledge (I did search), don't appears in the documentation (if you want to leave it like that, should be documented). Maybe there are other output formats that are not ready to process, because _some previous definition_ that should not be there in the 1st place.
